src/pythonfinder/models/python.py

```diff
@@ -392,19 +492,24 @@ def from_path(cls, path, name=None, ignore_unsupported=True):
         if not path.is_python:
             if not (ignore_unsupported or IGNORE_UNSUPPORTED):
                 raise ValueError("Not a valid python path: %s" % path.path)
-        py_version = get_python_version(path.path.absolute().as_posix())
-        instance_dict = cls.parse(py_version.strip())
+        try:
+            name = getattr(path, "name", path.path.name)
+            instance_dict = cls.parse(name)
```

Comment:
I think mypy is not smart here. Actually ```path.path.name``` will never to ```None```, and so as ```name```. But mypy thinks ```name``` could be ```None```. I think it can be safely ignored.

```
error: Argument 1 to "parse" of "PythonVersion" has incompatible type "Optional[str]"; expected "str"
```

Reply:
An explicit # type: str comment would fix that issue.